src/Modules/Settings/Dnn.PersonaBar.Security/Attributes/AuthProviderSupportedAttribute.cs

```diff
@@ -0,0 +1,52 @@
+﻿#region Copyright
+// 
+// DotNetNuke® - http://www.dotnetnuke.com
+// Copyright (c) 2002-2017
+// by DotNetNuke Corporation
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
+// of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+// DEALINGS IN THE SOFTWARE.
+#endregion
+
+using System;
+using System.Linq;
+using System.ComponentModel.DataAnnotations;
+using Dnn.PersonaBar.Security.Components;
+using DotNetNuke.Services.Localization;
+
+namespace Dnn.PersonaBar.Security.Attributes
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    class AuthProviderSupportedAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var propertyName = validationContext.DisplayName;
+
+            if (string.IsNullOrWhiteSpace(value.ToString()))
+            {
+                return new ValidationResult(string.Format(Localization.GetString(Constants.NotValid + ".Text", Constants.LocalResourcesFile), propertyName, value.ToString()));
+            }
+
+            var allAuthProviders = new SecurityController().GetAuthenticationProviders();
+
+            if (!allAuthProviders.Contains(value))
+            {
+                return new ValidationResult(string.Format(Localization.GetString(Constants.NotValid + ".Text", Constants.LocalResourcesFile), propertyName, value.ToString()));
```

Comment:
Please check this entry for resource file for localization

Reply:
no need to append ".Text", Localization.GetString() is able to get that

Reply:
can we get rid of the extra space between {0} and {1} in entry "{0}  {1} is not valid."?